src/crypto/crypto_ec.cc

```diff
@@ -314,7 +314,8 @@ void ECDH::SetPrivateKey(const FunctionCallbackInfo<Value>& args) {
     return THROW_ERR_CRYPTO_OPERATION_FAILED(env,
         "Failed to set generated public key");
 
-  EC_KEY_copy(ecdh->key_.get(), new_key.get());
+  // Use EC_KEY_dup instead of EC_KEY_copy for compatibility with BoringSSL.
+  ecdh->key_.reset(EC_KEY_dup(new_key.get()));
```

Comment:
Please add a `CHECK(ecdh->key_)` to ensure that the allocation was successful, otherwise, the next line will crash with a segmentation fault.

That being said...

Based on my understanding of this function, `new_key` was created by `EC_KEY_dup`'ing `ecdh->key_.get()`, presumably to avoid modifying the existing `ecdh->key_` in case of an error. With this patch, I think we are duplicating it again (which involves more allocations than copying the contents only).

`new_key` isn't needed after this function returns as far as I can tell. Can't we move `new_key` to `ecdh->key_` instead of duplicating it and then deallocating `new_key`?

Reply:
@tniessen done!